### PR TITLE
Add Code of Conduct

### PR DESCRIPTION
**Pull request by djhoese**
In an effort to make VisPy a more welcoming community, I'd like to add a Code of Conduct. A code of conduct allows members of the VisPy community to know what is expected of them and of the other members of the community. It also lets everyone know what actions are taken due to violations of this code of conduct regardless of the violator's "status" in the project. That means even core developers/maintainers are subject to this CoC and its punishments (including permanent ban from the community). This CoC applies to both virtual (GitHub, gitter chat, video calls) as well as in-person events (sprints, conferences, meetings, etc).

TODO:

- [x] Create an email list specifically for community members to submit violations of this CoC (new google group?  https://groups.io/?)
- [x] Identify who will be on this mailing list and can appropriately respond to violation submissions and enforce any punishment.
- [x] Add link to CoC document from vispy website
- [x] Add small description and link to repository README

I will be on the mailing list. Who else is willing?

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+an individual is officially representing the community in public spaces.
+Examples of representing our community include using an official e-mail address,
+posting via an official social media account, or acting as an appointed
+representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the community leaders responsible for enforcement at
+[INSERT CONTACT METHOD].
```
Comment (almarklein) [inline CODE_OF_CONDUCT.md:63]: todo

Comment (djhoese) [reply CODE_OF_CONDUCT.md:63]: That's what the first TODO item is for :wink:

**Review by rossant — APPROVED** [review]

**Comment by rossant** [comment]
> * Create an email list specifically for community members to submit violations of this CoC (new google group?  https://groups.io/?)

Would anyone have to join the mailing list prior to sending a violation?

Would it be possible to have an email instead, e.g. "conduct@vispy.org", that would redirect to a predefined set of persons?

**Review by kmuehlbauer — APPROVED** [review]

**Comment by almarklein** [comment]
> Would it be possible to have an email instead, e.g. "conduct@vispy.org", that would redirect to a predefined set of persons?

You'd need a server to receive and then forward the email. Not sure if there are reliable+free services to do so at a custom domain. We could do it with `vispy-conduct@gmail.com`. But if ppl can simply send an email to the mailing list without having to subscribe or anything, that might be easier to manage?

**Comment by rossant** [comment]
> You'd need a server to receive and then forward the email. Not sure if there are reliable+free services to do so at a custom domain.

It depends on the registrar. With namecheap, I know you can have email redirections, where `foo@domain.com` redirects to N existing email addresses.

> But if ppl can simply send an email to the mailing list without having to subscribe or anything, that might be easier to manage?

Possibly, yes.

**Comment by djhoese** [comment]
I know for groups.io you don't have to join the group to send an email to it. I don't remember about google groups. The one problem I had when setting up a groups.io list for another project was that you had to consciously make sure to CC the original emailer because they weren't included in the Reply All by default.

**Comment by djhoese** [comment]
@almarklein You're in charge of all of the vispy.org stuff, right? Any options for what @rossant is talking about?

**Comment by almarklein** [comment]
Yes, technically I still own the domain. It's registered at TransIP. They also have email, but not free/cheap. One option could be to transfer to NameCheap. Not sure if that's worth the hassle though; a dedicated gmail account or maillist seems simpler.

**Comment by djhoese** [comment]
I think groups.io is just easier and I've seen other projects use it. That said, we have the vispy-dev and vispy google groups...but I have an unfounded feeling that we are more likely to get spam on a google group that allows posts by anyone. groups.io is a little more mailing list-ish.

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (djhoese) [reply CODE_OF_CONDUCT.md:63]:
```suggestion
vispy-conduct@groups.io.
```

**Review by almarklein — APPROVED** [review]

**Comment by djhoese** [comment]
Ok, I've added a badge to the README that links to the file and I've updated the docs submodule to include the most recent documentation (which includes a link to the CoC). I'll merge this once the tests pass.

**Comment by almarklein** [comment]
This test seems to be rather flaky:
```
Example /basics/plotting/scatter_histogram.py failed (0):
[...]
numpy.linalg.LinAlgError: Singular matrix
```

**Comment by djhoese** [comment]
This PR is purely documentation and the website job passes so I'm going to merge this and we can work on the CI failures in other PRs/issues.

...I'm sick of seeing this being open.

**Pull request merged**